core/state/statedb.go

```diff
@@ -1243,55 +1251,59 @@ func (s *StateDB) commit(deleteEmptyObjects bool) (*stateUpdate, error) {
 
 	origin := s.originalRoot
 	s.originalRoot = root
-	return newStateUpdate(origin, root, deletes, updates, nodes), nil
+
+	s.commit = newStateUpdate(origin, root, deletes, updates, nodes)
+	return nil
 }
 
-// commitAndFlush is a wrapper of commit which also commits the state mutations
-// to the configured data stores.
-func (s *StateDB) commitAndFlush(block uint64, deleteEmptyObjects bool) (*stateUpdate, error) {
-	ret, err := s.commit(deleteEmptyObjects)
-	if err != nil {
-		return nil, err
+// Flush operates on a comitted-but-not-flushed state database to iterate and
+// push the state diffs into the database. Unless you need to do something in
+// between commit and flush, just use statedb.Commit directly.
+func (s *StateDB) Flush(block uint64) (common.Hash, error) {
```

Comment:
Spontanously, I think it would make sense for `Flush` to set `s.commit` to `nil`, so if two flushes in a row  is performed, the second is a no-op. That would also match with the "warn if overwrite s.commit" that I suggested earlier. 
Is there any reason why we'd want to retain the `commit` set after flush?

Reply:
I did do it like that, but the fuzzer wants access to the update struct, so we'd need to export it somehow else.

Reply:
What fuzzer is problematic?